tkeysign.go

```diff
@@ -271,24 +271,29 @@ func (s Signer) getSig() ([]byte, error) {
 	return rx[3 : 3+64], nil
 }
 
-// SignPh signs a SHA512 pre-hashed message in data and returns an
-// ed25519ph signature.
-func (s Signer) SignPh(data [64]byte) ([]byte, error) {
+// GetFWDigest asks the signer to hash len bytes of the firmware.
+//
+// It returns the resulting BLAKE2s digest or an error.
+func (s Signer) GetFWDigest(len int) ([]byte, error) {
 	id := 2
-	tx, err := tkeyclient.NewFrameBuf(cmdSignPhData, id)
+	tx, err := tkeyclient.NewFrameBuf(cmdGetFirmwareHash, id)
 	if err != nil {
 		return nil, fmt.Errorf("NewFrameBuf: %w", err)
 	}
 
-	copy(tx[2:], data[:])
+	// Set firmware length as 32 bit LE
+	tx[2] = byte(len)
+	tx[3] = byte(len >> 8)
+	tx[4] = byte(len >> 16)
+	tx[5] = byte(len >> 24)
 
-	tkeyclient.Dump("LoadSignPhData tx", tx)
+	tkeyclient.Dump("GetFirmwareHash tx", tx)
 	if err = s.tk.Write(tx); err != nil {
 		return nil, fmt.Errorf("Write: %w", err)
 	}
 
 	// Wait for reply
-	rx, _, err := s.tk.ReadFrame(rspSignPhData, id)
+	rx, _, err := s.tk.ReadFrame(rspGetFirmwareHash, id)
 	if err != nil {
```

Comment:
Potentially that one would want to have a dump of the rx content as well.
`tkeyclient.Dump("GetFirmwareHash rx", rx)`